### PR TITLE
Add a small example

### PR DESCRIPTION
**Pull request by mit-mit**

### CODE REVIEW

File: example/main.dart [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import 'package:http/http.dart' as http;
+import 'dart:convert' as convert;
```
Comment (natebosch) [inline example/main.dart]: [nit] put this import first with a blank line in between

https://www.dartlang.org/guides/language/effective-dart/style#do-place-dart-imports-before-other-imports

Comment (mit-mit) [reply example/main.dart]: Done

File: example/main.dart [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+main(List<String> arguments) {
+  // This example uses the Google Books API to search for books about http.
+  // https://developers.google.com/books/docs/overview
+  var url = "https://www.googleapis.com/books/v1/volumes?q={http}";
+
+  // Await the http get response, then decode the json-formatted responce.
```
[the remark this answers is not in the document]
Comment (mit-mit) [reply example/main.dart]: Done

[the remark this answers is not in the document]
Comment (mit-mit) [reply example/main.dart]: I tried to update `readme.md` also, but doesn't look like the code `client.get(response.bodyFields['uri'])` works anymore -- I don't see a `bodyFields` on `response`?

[the remark this answers is not in the document]
Comment (kevmoo) [reply example/main.dart]: 🤷‍♂️ – maybe we should sync the example in the readme w/ the one you're adding?

File: example/pubspec.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+name: httpexample
```
Comment (kevmoo) [inline example/pubspec.yaml]: This isn't necessary, right? Just drop the pubspec. The package site picks this up fine.

Comment (mit-mit) [reply example/pubspec.yaml]: How would you run the example then?

Comment (kevmoo) [reply example/pubspec.yaml]: `dart example/example.dart` 🤷‍♂️

Comment (mit-mit) [reply example/pubspec.yaml]:
```
$ ls -a
.		..		main.dart
$
```

Comment (mit-mit) [reply example/pubspec.yaml]: Never mind, my mistake, I had moved this out into another directory structure. Working now...

**Pull request merged**